RELEASE.md

```diff
@@ -0,0 +1,22 @@
+# Creating a release of mapbox.js
+
+### Before Deploying
+
+* You need to have the AWS CLI installed. `pip install awscli`
+* Write a `CHANGELOG.md` entry that describes your changes and links to
+  issues in question
+
+### Automated Deployement
+
+`npm run release <major.minor.patch>`
+
+### Manual Deployement
+
+* Update `_config.yaml`, `_config.publisher-production.yml`, `_config.publisher-staging.yml`
+* Build docs. `./deploy.sh v<major.minor.patch>`
```

Comment:
@whyvez the mapbox.js build should be published and working before docs are updated, right? Otherwise, the documentation will fail to load the latest mapbox.js release. If this is the case we should move the documentation update steps to its own separate section at the end. Looping in @katydecorah for guidance here.

Reply:
@alulsh I believe the actual issue was that the version in the `_config*.yml` files were bumped but the documentation pages were not generated. You can see this in the test added test here: https://github.com/mapbox/mapbox.js/pull/1309. I've also updated the script to make the config version and docs update in the same commit.

Reply:
Your explanation looks right to me, @whyvez. Updating the version number in `_config.yml` will cause the mapbox.js site to attempt to redirect to an `index.html` within a directly with the same name as the version number. So, if you tried to go to: 

```
https://www.mapbox.com/mapbox.js/
```

You would be redirected to 

```
https://docs.mapbox.com/mapbox.js/api/v3.2.1/
```

An ideal update would both 1) bump the version number and 2) include newly-generated docs for that specific version.

Reply:
Updated.